### ISSUE TITLE
Not sure if a bug or best practice? EventContent function issue

### ISSUE THREAD

**Issue by mtm81** [fullcalendar/fullcalendar#6396]
Using the timeline views, the length of the event visually is decided by the length of time the event is for.  If the title of the event is longer than that, the title is simply hidden.  A simply pop-up script can then be used to display it in full if needed.

So far, so good.

On previous versions of FullCalendar, if you wanted to further add things *such as html icons) into the title, you could use the EventRender function and add elements into the title.  All worked well.  

The issue comes with the "new" EventContent and EventDidMount functions.

If using the EventContent function, if you add elements into the title, the whole title is then displayed, the event grows in height and the ability for it to "self cut off" it's extended title seems to no longer happen.

If however, you use the EventDidMount function and effectively hack it to add elements into the title, because that happens AFTER it's finished mounting, the ability for the calendar to "self-hide" long titles doesn't seem to be affected.

The obvious solution would be to just use the EventDidMount function as above, however, when switching between views, because this isn't always fired, the element which were added into the title using the original call to this function are removed.

Example:
Lets say we have an event which is an hour long.  The title is extremely long.  PLUS we want to dynamically add some kind of glyph icons to the beginning of it.

1) If we don't do anything, then the event will  display as a single line, for the one hour duration with the title cut off visually.  (We can use a popup to give more info if needed.

2) If we use the EventDidMount to add the glyphs to the title - it's a bit of a hack, but the result is the same as 1) plus we get the icons we need.  Downside is - a change of view removes all of the work that's done on the EventDidMount first call and so we then lose our icons.

3) If we use the EventContent function, we get the glyphs as above, however the event then grows in height to display the full title rather than staying as a single line height with "auto hide" working.

 Essentially there is either an issue in the EventContent function or perhaps I'm missing something which would give me the result I need?

**Comment by Doc999tor** [comment]
@mtm81 Have you managed to solve this issue? I stuck with a similar problem

**Comment by mtm81** [comment]
Hi, Yes I did solve it in the end.
I had to use both of the methods to accomplish my task.  The main lightbulk moment was when using the EventContent function - obviously that replaced the ENTIRE contents of what would typically be generated by the script.

What I hadn't fully understood was that I needed to effectively 're-insert' elements within the event which were removed as part of my use of the EventContent function, but I still needed to allow the other parts still to work (such as the autohide of a long name etc)

Example is probably easier.

Lets say in your EventContent function you have a varaible called "thistitle".  That thistitle holds the event info you want to display in your event.

typically you would probably get that in place at the beginning of your function call, and then at the end, use something like;

var thistitle = "EVENTINFO HERE"

thisdomitem.innerHTML = thistitle
        let arrayOfDomNodes = [ thisdomitem ]
        return { domNodes: arrayOfDomNodes }


To fix my issue, I had to change things around, so that after setting the thistitle value, I then appended/prepended my prefferred value with some html which effectively 're-inserted' some of the tags which would be typically generated by the calendar code but is removed when setting it yourself, so my script ended up with something like:

var thistitle = "EVENTINFO HERE"

thistitle = '<div class="fc-event-main-frame"><div class="fc-event-title-container"><div class="fc-event-title fc-sticky">' + thistitle + "</div></div></div>"

thisdomitem.innerHTML = thistitle
        let arrayOfDomNodes = [ thisdomitem ]
        return { domNodes: arrayOfDomNodes }



Now, you requirements may well be different, however the way I figured this out was to compare the html generated normally (using the developer viewer in my browser), with the html generated when I used the eventcontent function and inserted a very simple title.  The different between to the two (in terms of the "missing" html) was the stuff I needed to manually put back in.


Hope this helps
